runtime/src/main/java/org/corfudb/runtime/LogReplicationUtils.java

```diff
@@ -42,120 +59,211 @@ public final class LogReplicationUtils {
 
     private LogReplicationUtils() { }
 
-    public static void subscribe(@Nonnull LogReplicationListener clientListener, @Nonnull String namespace,
-                          @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest, int bufferSize,
-                          CorfuStore corfuStore) {
+    public static void subscribe(@Nonnull LogReplicationListener listener, @Nonnull String namespace,
+                                 @Nonnull String streamTag, @Nonnull List<String> tablesOfInterest, int bufferSize,
+                                 CorfuStore corfuStore, ExecutorService clientExecutorService) {
+        log.info("Client subscription process has started.");
+
+        Map<String, List<String>> nsToTableName = new HashMap<>();
+        nsToTableName.put(namespace, tablesOfInterest);
+        nsToTableName.put(CORFU_SYSTEM_NAMESPACE, Arrays.asList(REPLICATION_STATUS_TABLE_NAME));
+
+        Map<String, String> nsToStreamTags = new HashMap<>();
+        nsToStreamTags.put(namespace, streamTag);
+        nsToStreamTags.put(CORFU_SYSTEM_NAMESPACE, LR_STATUS_STREAM_TAG);
+
```

Comment:
Will have to check this condition also - https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/runtime/collections/streaming/StreamPollingScheduler.java#L167